<launch>
  <arg name="input_cloud"
       default="/head_mount_kinect/hd/republished_points"/>
  <arg name="input_image"
       default="/head_mount_kinect/hd/image_color_rect"/>
  <arg name="input_camera_info"
       default="/head_mount_kinect/hd/camera_info"/>

  <arg name="input_rects" default="/qatm/output/original_size_rects" />
  <arg name="input_labels" default="/qatm/output/labels" />

  <node name="rect2label_array"
        pkg="what_i_see" type="rect2labelarray.py"
        output="screen">
    <remap from="~input/rect" to="$(arg input_rects)" />
    <remap from="~input/label" to="$(arg input_labels)" />
    <rosparam subst_value="true">
      approximate_sync: true
      convert_type: label
    </rosparam>
  </node>

  <node name="rect_projector"
        pkg="what_i_see" type="ssd_projector"
        output="screen">
    <param name="lazy_sync" value="true"/>
    <remap from="~input_image" to="$(arg input_image)"/>
    <remap from="~input_cloud" to="$(arg input_cloud)"/>
    <remap from="~input_info"  to="$(arg input_camera_info)"/>
    <remap from="~input_boxes" to="rect2label_array/output/boxes"/>
  </node>

</launch>
